telemetry/telemetry_record.py

```diff
@@ -73,15 +100,48 @@ def move_wpr_files(name_schema):
     """
     remote_data_path = os.path.join(CHROMIUM_SRC, 'tools/perf/page_sets/data/',
             name_schema)
-    local_path = os.path.join(os.getcwd(), 'data/')
+    local_path = os.path.join(os.getcwd(), 'data/wpr_source')
     p = Popen('mv {0} {1}'.format(remote_data_path, local_path), shell=True)
 
+def make_modified_json(num_urls):
+    """Creates *.json files to point to the right archive files
+
+    Located in ~/src/tools/perf/page_sets/data
+    :param num_urls: int for the number of target urls
+    TODO: add sha1?
+    """
+    json_path = '/home/jamshed/src/tools/perf/page_sets/data/'
+    json_template = ('{{'
+            '"description": "Describes the Web Page Replay archives for a user '
+            'story set. Dont edit by hand! Use record_wpr for updating.", '
+            '"archives": {{ "url{0}_page_set_000.wpr": ["url{0}"]}}}}\n')
+    for i in range(num_urls):
+        modified_name = '999999' + str(i)
+        file_name = 'url{0}_page_set.json'
+        with open('{0}{1}'.format(json_path,
+            file_name.format(modified_name)), 'wb') as f:
+            f.write(json_template.format(modified_name))
+
 def modify_wpr():
-    """Sets cacheable objects delay time to 0, creates  pc.wpr files"""
-    wpr_directory = os.path.join(os.getcwd(), 'data/')
+    """Sets cacheable objects delay time to 0, creates pc.wpr files
+
+    Note: *.pc.wpr files are converted to 999999 + str(url name), which lets
+    them be treated like regular urls. Just know that six 9's indicate a
+    modified wpr file
+    TODO: Fix the 999999 notation
```

Comment:
Hm, that's a pain!

Random thought: while we do need the results files to have the URL in the
filename, do we really need the benchmark to have the URL in the filename?
Could we just create files like test1.py, test2.py, test3.py... for the
benchmarks, but then in-memory, keep the URL for the current benchmark,
then write out the results to <base64 url>.db

On Fri, Apr 17, 2015 at 10:04 PM, Jamshed Vesuna <notifications@github.com>
wrote:

> In telemetry/telemetry_record.py
> <https://github.com/colin-scott/page_load_time/pull/2#discussion_r28641444>
> :
>
> >  def modify_wpr():
> > -    """Sets cacheable objects delay time to 0, creates  pc.wpr files"""
> > -    wpr_directory = os.path.join(os.getcwd(), 'data/')
> > +    """Sets cacheable objects delay time to 0, creates pc.wpr files
> > +
> > +    Note: *.pc.wpr files are converted to 999999 + str(url name), which lets
> > +    them be treated like regular urls. Just know that six 9's indicate a
> > +    modified wpr file
> > +    TODO: Fix the 999999 notation
>
> This looks really good. I noticed that base64 pads the output with ='s.
> As this is not allowed in python class names, I could have an intermediary
> function that checks the length and adds or removes the padding. Then I can
> use the above. Does that work?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/colin-scott/page_load_time/pull/2/files#r28641444>.
>